ldes-fragmentisers/ldes-fragmentisers-timebased/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/timebased/services/OpenFragmentProvider.java

```diff
@@ -2,29 +2,43 @@
 
 import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragment.entities.LdesFragment;
 import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragment.repository.LdesFragmentRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.timebased.config.TimebasedFragmentationConfig;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
 
 public class OpenFragmentProvider {
 
 	private final TimeBasedFragmentCreator fragmentCreator;
 	private final LdesFragmentRepository ldesFragmentRepository;
+	private final TimebasedFragmentationConfig timebasedFragmentationConfig;
 
 	public OpenFragmentProvider(TimeBasedFragmentCreator fragmentCreator,
-			LdesFragmentRepository ldesFragmentRepository) {
+			LdesFragmentRepository ldesFragmentRepository, TimebasedFragmentationConfig timebasedFragmentationConfig) {
 		this.fragmentCreator = fragmentCreator;
 		this.ldesFragmentRepository = ldesFragmentRepository;
+		this.timebasedFragmentationConfig = timebasedFragmentationConfig;
 	}
 
-	public LdesFragment retrieveOpenFragmentOrCreateNewFragment(LdesFragment parentFragment) {
+	public Pair<LdesFragment, Boolean> retrieveOpenFragmentOrCreateNewFragment(LdesFragment parentFragment) {
```

Comment:
Good fix to use a Pair here, but it's completely unclear what the boolean means unless you visit this class. Maybe abstract this in a class ?